src/diagnostic/DiagnosticOrigin.ts

```diff
@@ -0,0 +1,33 @@
+enum DiagnosticOrigin {
```

Comment:
Is there any reason we couldn't just use an object literal here? I have a preference for them over `enum`s unless there's a compelling reason 😉

Reply:
:+1: